core/src/main/java/org/apache/iceberg/view/ViewMetadataParser.java

```diff
@@ -132,16 +133,16 @@ public static ViewMetadata fromJson(JsonNode json) {
       historyEntries.add(ViewHistoryEntryParser.fromJson(vLog));
     }
 
-    return ImmutableViewMetadata.builder()
-        .location(location)
-        .currentVersionId(currentVersionId)
-        .properties(properties)
-        .versions(versions)
-        .schemas(schemas)
-        .currentSchemaId(currentSchemaId)
-        .history(historyEntries)
-        .formatVersion(formatVersion)
-        .build();
+    return ImmutableViewMetadata.of(
```

Comment:
switching this part so that we don't call `builder().upgradeFormatVersion()` when just setting the value